fault-proof/tests/sync.rs

```diff
@@ -801,4 +804,215 @@ mod sync {
+    ///
+    /// Verifies that the flag is correctly set based on:
+    /// - Game status (IN_PROGRESS vs resolved)
+    /// - Deadline status (passed vs not passed)
+    /// - Parent resolution status (resolved vs IN_PROGRESS)
+    /// - Ownership (own game vs not own game)
+    #[tokio::test]
+    async fn test_in_progress_games_resolution_marking() -> Result<()> {
+        let (env, proposer, init_bond) = setup().await?;
+
+        let starting_l2_block = env.anvil.starting_l2_block_number;
+
+        // Create a simple chain: M -> 0 -> 1 -> 2 -> 3 -> 4
+        // Add 100-second gaps between creations to control deadline spacing
+        let mut game_addresses = Vec::new();
+        let time_gap = 100u64;
+
+        let total_games = 5;
+        let num_time_gaps = total_games - 1;
+
+        for i in 0..total_games {
+            let block = starting_l2_block + 1 + i;
+            let root_claim = env.compute_output_root_at_block(block).await?;
+            let parent_id = if i == 0 { M } else { (i - 1) as u32 };
+            env.create_game(root_claim, block, parent_id, init_bond).await?;
+            let (_, address) = env.last_game_info().await?;
+            game_addresses.push(address);
+            tracing::info!("✓ Created game {i} at block {block} with parent {parent_id}");
+
+            // Add time gap before next game (except after last game)
+            if i < num_time_gaps {
+                env.warp_time(time_gap).await?;
+            }
+        }
+
+        // Sync state to cache games
+        proposer.sync_state().await?;
+
+        // === Case 1: IN_PROGRESS game with deadline NOT passed ===
+        let game_0 = proposer.get_game(U256::from(0)).await.unwrap();
+        assert!(
+            !game_0.should_attempt_to_resolve,
+            "Game 0: should_attempt_to_resolve should be false (deadline not passed)"
+        );
+        tracing::info!("✓ Case 1: IN_PROGRESS + deadline not passed → should_attempt = false");
+
+        // === Case 2: DEFENDER_WINS (resolved game) ===
+        // Warp to just past game 0's deadline but before game 1's deadline
+        // Game 0 deadline is at T0 + MAX_CHALLENGE_DURATION
+        // Game 1 deadline is at T0 + 100 + MAX_CHALLENGE_DURATION
+        // Currently at T0 + 400 (after creating all `total_games` games with `time_gap`-second
+        // gaps)
+        // So we need to warp: MAX_CHALLENGE_DURATION - `num_time_gaps` * `time_gap` + 1
+        env.warp_time(MAX_CHALLENGE_DURATION - num_time_gaps * time_gap + 1).await?;
+
+        proposer.sync_state().await?;
+        let game_0 = proposer.get_game(U256::from(0)).await.unwrap();
+        assert!(
+            game_0.should_attempt_to_resolve,
+            "Game 0: should_attempt_to_resolve should be true since deadline has passed"
+        );
+
+        env.resolve_game(game_addresses[0]).await?;
+        proposer.sync_state().await?;
+
+        let game_0 = proposer.get_game(U256::from(0)).await.unwrap();
```

Comment:
```suggestion
        let game_0 = proposer.get_game(U256::from(0)).await?;
```

Could you propogate with `?` in other places too?